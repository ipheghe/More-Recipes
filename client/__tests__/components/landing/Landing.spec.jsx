import expect from 'expect';
import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import render from 'react-test-renderer';
import ConnectedLandingPage, { PureLanding }
  from '../../../src/components/Landing';
import mockItems from '../../__mocks__/mockItems';
import mockAuthCheck from '../../__mocks__/mockAuthCheck';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const initialState = {
  auth: {
    error: 'Invalid username',
    message: '',
    userData: {},
    authenticated: false,
    categories: []
  },
  recipe: {
    message: '',
    error: '',
    pages: 2,
    recipeData: {},
    recipeList: [],
    userRecipes: [],
    searchResult: []
  }
};

const store = mockStore(initialState);

const props = {
  recipeList: mockItems.recipeArray,
  getTopRecipesLanding: jest.fn(() => Promise.resolve()),
  isAuthenticated: false
};

/**
 *@description  setup function to mount component
 *
 * @param { boolean } isAuthenticated
 *
 * @return { * } null
 */
const setup = (isAuthenticated) => {
  props.isAuthenticated = isAuthenticated;
  const mountedWrapper =
  mount(<Router><ConnectedLandingPage {...props} store={store} /></Router>);
  const shallowWrapper = shallow(<PureLanding {...props} />);
  return {
    mountedWrapper,
    shallowWrapper
  };
};

describe('<Landing', () => {
  beforeEach(() => {
    mockAuthCheck();
  });

  it('renders without crashing', () => {
    const { mountedWrapper } = setup(false);
    expect(mountedWrapper.find('LandingRecipeList').length).toBe(1);
    expect(mountedWrapper.exists()).toBe(true);
  });

  it('calls componentDidMount', () => {
    sinon.spy(PureLanding.prototype, 'componentDidMount');
    const { shallowWrapper } = setup(false);
    expect(shallowWrapper.exists()).toBe(true);
    expect(PureLanding.prototype.componentDidMount.calledOnce).toEqual(false);
  });

  it(`calls showTopRecipes event when 
     view top recipes button is clicked`, () => {
      const { shallowWrapper } = setup(false);
      const event = {
        preventDefault: jest.fn()
      };
      shallowWrapper.instance().showTopRecipies(event);
    });

  it('should match component snapshot', () => {
    const tree = render.create(<Router ><PureLanding {...props} /></Router>);
    expect(tree).toMatchSnapshot();
  });

  it('redirects to dashboard page if user is authenticated', () => {
    const { shallowWrapper } = setup(true);
    expect(shallowWrapper.instance().props.isAuthenticated).toEqual(true);
  });
});
