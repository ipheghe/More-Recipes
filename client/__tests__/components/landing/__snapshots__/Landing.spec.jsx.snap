// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing should match component snapshot 1`] = `
<div>
  <div
    className="landing-background"
    id="landing-background"
  >
    <div
      className="container"
    >
      <div
        className="row landing"
      >
        <section
          className="col-md-7 headline-home"
        >
          <h1>
            Welcome to More recipes
          </h1>
          <h4>
            Share your recipe ideas with the world
          </h4>
          <br />
          <br />
          <button
            className="btn btn-success"
            onClick={[Function]}
            style={
              Object {
                "width": "300px",
              }
            }
          >
            View top recipes
          </button>
          <p
            className="arrow-key"
          >
            <i
              aria-hidden="true"
              className="fa fa-arrow-down"
            />
            <i
              aria-hidden="true"
              className="fa fa-arrow-down"
            />
            <i
              aria-hidden="true"
              className="fa fa-arrow-down"
            />
          </p>
        </section>
        <section
          className="col-md-5 account"
        >
          <div>
            <a
              className="btn btn-success home"
              href="#/login"
              id="login"
              onClick={[Function]}
            >
              Login
            </a>
            <p
              className="brief"
            >
              Login to your account and start sharing
            </p>
          </div>
          <div>
            <a
              className="btn btn-success home"
              href="#/signup"
              id="signup"
              onClick={[Function]}
            >
              Signup
            </a>
            <p
              className="brief"
            >
              Register now to enjoy all of More recipes
            </p>
          </div>
        </section>
      </div>
    </div>
  </div>
  <section
    className="top-recipes"
    id="top-recipes"
  >
    <div
      className="heading"
    >
      <h3>
        Welcome to the top recipes of the week
      </h3>
    </div>
    <div
      className="card-blocks-home"
    >
      <div>
        <div
          className="card"
        >
          <img
            alt="recipe"
            className="card-img-top"
            src="assets/images/pizza1.jpg"
          />
          <div
            className="card-block"
          >
            <h4
              className="card-title"
            >
              Banga Soup
            </h4>
            <p
              className="card-text"
            >
              This recipe is very popular in the south south part of Nigeria
            </p>
          </div>
          <div
            className="card-footer"
          >
            <small
              className="text-muted"
            >
              <div
                className="landing-card-footer"
              >
                <span
                  className="views"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-eye"
                    style={
                      Object {
                        "color": "green",
                      }
                    }
                  />
                </span>
                <span
                  className="upvote"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-thumbs-up"
                    style={
                      Object {
                        "color": "red",
                      }
                    }
                  />
                </span>
                <span
                  className="downvote"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-thumbs-down"
                    style={
                      Object {
                        "color": "orange",
                      }
                    }
                  />
                </span>
              </div>
            </small>
          </div>
        </div>
      </div>
    </div>
  </section>
  <br />
</div>
`;
